static/app/views/issueList/issueListSetAsDefault.spec.tsx

```diff
@@ -34,34 +42,35 @@ describe('IssueListSetAsDefault', () => {
       }),
     });
 
-    render(<IssueListSetAsDefault {...defaultProps} />);
+    render(<IssueListSetAsDefault {...defaultProps} />, {organization});
 
     userEvent.click(screen.getByRole('button', {name: /set as default/i}));
 
     await waitFor(() => {
       expect(mockPinSearch).toHaveBeenCalledWith(
         expect.anything(),
         expect.objectContaining({
-          data: {query: 'is:unresolved', sort: 'date', type: SavedSearchType.ISSUE},
+          data: {query: 'browser:firefox', sort: 'date', type: SavedSearchType.ISSUE},
         })
       );
     });
   });
 
   it('can remove a default search', async () => {
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/searches/',
+      body: [TestStubs.Search({isPinned: true, query: 'browser:firefox'})],
+    });
     const mockUnpinSearch = MockApiClient.addMockResponse({
       url: '/organizations/org-slug/pinned-searches/',
       method: 'DELETE',
     });
 
-    render(
-      <IssueListSetAsDefault
-        {...defaultProps}
-        savedSearch={TestStubs.Search({isPinned: true})}
-      />
-    );
+    render(<IssueListSetAsDefault {...defaultProps} />, {organization});
 
-    userEvent.click(screen.getByRole('button', {name: /remove default/i}));
+    await waitFor(() => {
+      userEvent.click(screen.getByRole('button', {name: /remove default/i}));
+    });
```

Comment:
Why do you need this? does can't you just use something like 

```suggestion
    userEvent.click(await screen.findByRole('button', {name: /remove default/i}));
```

Reply:
Or do we have to wait for the button to become enabled first or something

Reply:
Nope, the way you wrote it works and is better, don't remember why I had it like this